imap/imapd.c

```diff
@@ -15866,6 +15868,31 @@ static void cmd_notify(char *tag, int set)
                 goto missingclose;
             }
 
+            /* Sanity check events.  Per RFC 5465, Section 5:
+
+               If the FlagChange and/or AnnotationChange events are specified,
+               MessageNew and MessageExpunge MUST also be specified by the client.
+               Otherwise, the server MUST respond with the tagged BAD response.
+
+               If one of MessageNew or MessageExpunge is specified, then both events
+               MUST be specified.  Otherwise, the server MUST respond with the
+               tagged BAD response.
+            */
+            if (((*events & (IMAP_NOTIFY_FLAG_CHANGE|IMAP_NOTIFY_ANNOTATION_CHANGE))
```

Comment:
I had to use pen and paper to validate that the comment matches the conditional. My source of confusion was if the "events" in the sentence "If one of MessageNew or MessageExpunge is specified, then both events" refer to MessageNew/MessageExpunge or rather to the FlagChange/AnnotationChange events in the paragraph before.
I know that this verbatim text from the RFC and take it that you know from IETF meetings what to make of this sentence better than me.

Reply:
As I read it, the only legal combinations of these 3 events are the following (which is what I have implemented):

[ MessageNew MessageExpunge [ FlagChange ] ]

Reply:
I agree, both in that this is more likely the intend of the specification. And that it is implemented correctly.